tracker/gioui/unit_editor.go

```diff
@@ -332,7 +332,6 @@ func (p ParameterStyle) Layout(gtx C) D {
 				gtx.Constraints.Min.Y = gtx.Dp(unit.Dp(40))
 				instrItems := make([]MenuItem, p.tracker.Instruments().Count())
 				for i := range instrItems {
-					i := i
```

Comment:
lol

Reply:
It's not a lol, it's a bug fix. But you are right, Golang 1.22 fixed this, so it's not needed anymore. See: https://go.dev/blog/loopvar-preview

Reply:
Yeah, it's a bit silly, that's why golang people changed the language semantics. Interestingly, same mistake in language design was made by C#, see https://learn.microsoft.com/en-us/archive/blogs/ericlippert/closing-over-the-loop-variable-considered-harmful. And they also changed the semantics of the language so you don't have to do `var v2 = v` type of assignments.

But so far, this has been the _only_ breaking change in v1 of go; disregarding this, all old go code since v1 still compiles & should work.

Why language designers make this mistake? My guess is that it's because at early stages of their language design, their compiler is not very good at optimizing, and `var v2 = v` or `i := i` (which actually is exactly identical i.e. defining a new variable, the name just overlaps the old name) both are NOT nops, but do some unnecessary memory / register assignments. Just a guess.